src/transformers/trainer_callback.py

```diff
@@ -403,11 +403,7 @@ def on_step_end(self, args: TrainingArguments, state: TrainerState, control: Tra
         # Log
         if state.global_step == 1 and args.logging_first_step:
             control.should_log = True
-        if (
-            args.logging_strategy == IntervalStrategy.STEPS
-            and args.logging_steps > 0
```

Comment:
note, I moved this check into the args init, symmetical to `eval_steps`, it's a hard check now, rather than a silent skip.

Reply:
This is potentially not BC, if someone used an explicit `--logging_steps 0` and logging strategy different from `no`.

So I'd be happy to revert this part of the change if you think it's a problem.

Reply:
It's fine here I think.